packages/admin-ui/src/text-area/text-area.tsx

```diff
@@ -0,0 +1,136 @@
+import type { ReactNode, ComponentPropsWithoutRef, Ref } from 'react'
+import React, { useRef, useState, forwardRef } from 'react'
+import { useId } from '@vtex/admin-ui-hooks'
+
+import { TextAreaContainer } from './text-area-container'
+import { TextAreaElement } from './text-area-element'
+import { Text } from '../components/Text'
+
+import {
+  FormControl,
+  FormControlLabel,
+  FormControlMessage,
+} from '../form-control'
+import { Flex, FlexSpacer } from '..'
+import { MAX_HEIGHT } from './text-area.style'
+
+export const TextArea = forwardRef(
```

Comment:
I think this component can benefit from the `prop-getters` pattern. You could basically create a hook that is responsible for returning the correct props for your textarea.

```jsx
function useTextarea(){
  // add the counter, line height and scroll things
  // ...
  
  // this method returns your textarea props
 const  getTextareaProps(customProps: React.ComponentPropsWithoutRef<'textarea'>) {
    return { 
      // ... return all relevant props for the component such as onInput, onChange
      // ... your custom props like id, disabled
    }
  }
}
```

The usage is pretty straightforward

```jsx
function Textarea(){
  const { getTextareaProps } = useTextarea()
  return <textarea {...getTextareaProps({ id, disabled, onChange, })}
}
```